sdk/data/azcosmos/cosmos_client.go

```diff
@@ -42,10 +42,11 @@ func (c *Client) Endpoint() string {
 // options - Optional Cosmos client options.  Pass nil to accept default values.
 func NewClientWithKey(endpoint string, cred KeyCredential, o *ClientOptions) (*Client, error) {
 	preferredRegions := []string{}
+	enableEndpointDiscovery := true
```

Comment:
nit: lets rename this, `enableEndpointDiscovery` melts the mind of anyone trying to understand